### PR TITLE
MySqlGroupChangeRepository

### LINKED ISSUE #173: MySQL Group Change Repository
Refer to https://github.com/vinyldns/vinyldns/issues/16

1. Create MySQLGroupChangeRepository that implements functionality of DynamoDBGroupChangeRepository
1. Add unit and integration tests

### PR DESCRIPTION
**Pull request by nimaeskandary**
Fixes #173 .

Changes in this pull request:
- adds myslq group change repo
- removes invalid repos check

### CODE REVIEW

File: modules/mysql/src/main/resources/db/migration/V3.7__GroupChanges.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+CREATE SCHEMA IF NOT EXISTS ${dbName};
+
+USE ${dbName};
+
+/*
+Create table to store group changes
+*/
+CREATE TABLE group_change (
+  group_change_id CHAR(36) NOT NULL,
```
Comment (pauljamescleary) [inline modules/mysql/src/main/resources/db/migration/V3.7__GroupChanges.sql]: we use `id`, not `group_change_id`

File: modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlGroupChangeRepository.scala [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import vinyldns.proto.VinylDNSProto
+
+class MySqlGroupChangeRepository extends GroupChangeRepository with Monitored {
+  import MySqlGroupChangeRepository._
+
+  private final val logger = LoggerFactory.getLogger(classOf[MySqlGroupChangeRepository])
+
+  private final val PUT_GROUP_CHANGE =
+    sql"""
+      |REPLACE INTO group_change (group_change_id, group_id, created_timestamp, data)
```
Comment (pauljamescleary) [inline modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlGroupChangeRepository.scala]: we should be using `INSERT INTO ON DUPLICATE KEY UPDATE` now right?

Comment (nimaeskandary) [reply modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlGroupChangeRepository.scala]: Ah right, forgot to make the changes i made in the membership repo here, could be insert ignore as well since im assuming group changes cant be altered?

File: modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlGroupChangeRepository.scala [hunk 3]
```diff
@@ -31,15 +31,15 @@ class MySqlGroupChangeRepository extends GroupChangeRepository with Monitored {
 
   private final val PUT_GROUP_CHANGE =
     sql"""
-      |REPLACE INTO group_change (group_change_id, group_id, created_timestamp, data)
+      |INSERT INTO ON DUPLICATE KEY UPDATE group_change (id, group_id, created_timestamp, data)
```
Comment (pauljamescleary) [inline modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlGroupChangeRepository.scala]: this does not appear to be correct?

The syntax should be...

```
INSERT INTO table_name(c1)
VALUES(c1)
ON DUPLICATE KEY UPDATE c1 = VALUES(c1) + 1;
```

Comment (pauljamescleary) [reply modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlGroupChangeRepository.scala]: Yea, the syntax looks wrong.  Also, make sure you have an integration test that tests the ON DUPLICATE KEY UPDATE part

File: modules/mysql/src/main/resources/db/migration/V3.7__GroupChanges.sql [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+/*
+Create table to store group changes
+*/
+CREATE TABLE group_change (
+  id CHAR(36) NOT NULL,
+  group_id CHAR(36) NOT NULL,
+  created_timestamp BIGINT(13) NOT NULL,
+  data BLOB NOT NULL,
+  PRIMARY KEY (id),
+  INDEX group_id_index (group_id)
```
Comment (mitruly) [inline modules/mysql/src/main/resources/db/migration/V3.7__GroupChanges.sql]: Kind of related/just wondering - should we create a card to update the index for `DyanmoDBGroupChangeRepository` to just `GROUP_ID` as well? If group ID will always be unique, not sure that we need a created timestamp.

Comment (nimaeskandary) [reply modules/mysql/src/main/resources/db/migration/V3.7__GroupChanges.sql]: Need the timestamp for paging

File: modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlGroupChangeRepository.scala (hunk 3 above)
Comment (nimaeskandary) [reply modules/mysql/src/main/scala/vinyldns/mysql/repository/MySqlGroupChangeRepository.scala]: Whoops, will update
